src/main/java/com/winner_is_kungen/tda367/model/Signal.java

```diff
@@ -0,0 +1,18 @@
+package com.winner_is_kungen.tda367.model;
+
+import com.winner_is_kungen.tda367.model.util.Tuple;
+
+import java.util.ArrayList;
+
+class Signal extends ArrayList<Tuple<ComponentListener,Integer,Integer>> {

```

Comment:
> Inheritance should make sense

`Signal` should not inherit `ArrayList` just because it should have a lot of the same methods unless it also actually is an `ArrayList`, Which it's not in our case.  
Use composition instead.